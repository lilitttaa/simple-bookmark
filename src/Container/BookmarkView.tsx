import { BookmarkItem } from './HomeContainer'

export function BookmarkViewContainer ({
  bookmarkItems,
  deleteBookmark
}: {
  bookmarkItems: BookmarkItem[]
  deleteBookmark: (id: string) => void
}) {
  return (
    <div className='BoomarkViewContainer'>
      <ul>
        {bookmarkItems.map(item => (
          <div className='BookmarkViewItem' key={item.id}>
            <li
              key={item.id}
              style={{
                display: 'inline'
              }}
            >
              <a href={item.url}>{item.title}</a>
            </li>

            <button
              onClick={() => {
                deleteBookmark(item.id)
              }}
              style={{
                display: 'inline'
              }}
            >
              Delete
            </button>
          </div>
        ))}
      </ul>
    </div>
  )
}
