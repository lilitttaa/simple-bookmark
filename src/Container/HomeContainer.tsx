import { useEffect, useState } from 'react'
import { BookmarkViewContainer } from './BookmarkView'

export type BookmarkItem = {
	id: string
	title: string
	url: string
  }
  const Home: React.FC = () => {
	const [title, setTitle] = useState('')
	const [url, setUrl] = useState('')
  
	const [bookmarkItems, setBookmarkItems] = useState([
	  {
		id: '1',
		title: 'Google',
		url: 'https://www.google.com'
	  },
	  {
		id: '2',
		title: 'Yahoo',
		url: 'https://www.yahoo.com'
	  }
	] as BookmarkItem[])
  
	const addBookmark = () => {
	  const newBookmarkItem = [...bookmarkItems]
	  newBookmarkItem.push({
		id: String(bookmarkItems.length + 1),
		title,
		url
	  })
	  setBookmarkItems(newBookmarkItem)
	}
	const deleteBookmark = (id: string) => {
	  const newBookmarkItem = bookmarkItems.filter(bookmark => bookmark.id !== id)
	  setBookmarkItems(newBookmarkItem)
	}
	useEffect(() => {}, [])
	return (
	  <div className='HomeContainer'>
		<div className='AddBookmarkContainer'>
		  <input
			type='text'
			placeholder='Title'
			onChange={e => setTitle(e.target.value)}
			value={title}
		  />
		  <input
			type='text'
			placeholder='URL'
			onChange={e => setUrl(e.target.value)}
			value={url}
		  />
		  <button
			onClick={() => {
			  addBookmark()
			  setTitle('')
			  setUrl('')
			}}
		  >
			Add
		  </button>
		</div>
		<BookmarkViewContainer
		  bookmarkItems={bookmarkItems}
		  deleteBookmark={deleteBookmark}
		></BookmarkViewContainer>
	  </div>
	)
  }
  
  export default Home
  