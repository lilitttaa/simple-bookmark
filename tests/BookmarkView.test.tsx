import renderer from 'react-test-renderer'
import { it, expect, afterEach } from '@jest/globals'
import { BookmarkViewContainer } from '../src/Container/BookmarkView'
import { cleanup, fireEvent, render } from '@testing-library/react'

afterEach(cleanup)

it('renders correctly', () => {
  const tree = renderer
    .create(
      <BookmarkViewContainer
        bookmarkItems={[
          {
            id: '1',
            title: 'Google',
            url: 'https://www.google.com'
          },
          {
            id: '2',
            title: 'Yahoo',
            url: 'https://www.yahoo.com'
          }
        ]}
        deleteBookmark={() => {}}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('click delete button', () => {
  const { queryByLabelText, getByLabelText,getByText } = render(
    <BookmarkViewContainer
      bookmarkItems={[
        {
          id: '1',
          title: 'Google',
          url: 'https://www.google.com'
        }
      ]}
      deleteBookmark={() => {}}
    />
  )
  const deleteButton = getByText('Delete')
  fireEvent.click(deleteButton)
  expect(queryByLabelText('Google')).toBeNull()
})
