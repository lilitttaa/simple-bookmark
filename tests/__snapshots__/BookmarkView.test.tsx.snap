// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="BoomarkViewContainer"
>
  <ul>
    <div
      className="BookmarkViewItem"
    >
      <li
        style={
          {
            "display": "inline",
          }
        }
      >
        <a
          href="https://www.google.com"
        >
          Google
        </a>
      </li>
      <button
        onClick={[Function]}
        style={
          {
            "display": "inline",
          }
        }
      >
        Delete
      </button>
    </div>
    <div
      className="BookmarkViewItem"
    >
      <li
        style={
          {
            "display": "inline",
          }
        }
      >
        <a
          href="https://www.yahoo.com"
        >
          Yahoo
        </a>
      </li>
      <button
        onClick={[Function]}
        style={
          {
            "display": "inline",
          }
        }
      >
        Delete
      </button>
    </div>
  </ul>
</div>
`;
